### PR TITLE
Added dataloader for Panoptic Mask R-CNN

### PR DESCRIPTION
**Pull request by srihari-humbarwadi**
# Description
Dataloader for Panoptic Mask R-CNN input

## Type of change
- [x] A new research paper code implementation

## Tests
`python -m official.vision.beta.projects.panoptic_maskrcnn.dataloaders.panoptic_maskrcnn_input_test`

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py [hunk 1]
```diff
@@ -15,10 +15,21 @@
...
 import dataclasses
-
 from official.vision.beta.configs import maskrcnn
 from official.vision.beta.configs import semantic_segmentation
 
+# pylint: disable=missing-class-docstring
+
+@dataclasses.dataclass
+class Parser(maskrcnn.Parser):
+  resize_eval_segmentation_groundtruth = True,
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py:25]: Can you add type annotation (i.e bool, int, Optional[List[int]], ... etc).

File: official/vision/beta/projects/panoptic_maskrcnn/dataloaders/panoptic_maskrcnn_input.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        gt_masks: Groundtruth masks cropped by the bounding box and
+          resized to a fixed size determined by mask_crop_size.
+        gt_segmentation_mask: Groundtruth mask for segmentation head, this is
+          resized to a fixed size determined by output_size.
+        gt_segmentation_valid_mask: Binary mask that marks the pixels that
+          are supposed to be used in computing the segmentation loss while
+          training.
+    """
+    segmentation_mask = data['groundtruth_segmentation_mask']
+    segmentation_mask = tf.expand_dims(segmentation_mask, axis=0)
```
Comment (arashwan) [inline official/vision/beta/projects/panoptic_maskrcnn/dataloaders/panoptic_maskrcnn_input.py:154]: We usually assume the channel dimension is the -1 dimension. Can you double check preprocess_ops.random_horizontal_flip handles this correctly?

Comment (srihari-humbarwadi) [reply official/vision/beta/projects/panoptic_maskrcnn/dataloaders/panoptic_maskrcnn_input.py:154]: https://github.com/tensorflow/models/blob/2e9c6046662658f3e3d3e31ebfebdc4481185c75/official/vision/beta/ops/preprocess_ops.py#L530-L557

IIUC, `horizontal_flip_masks` expects masks of shape [N, H, W], and `segmentation_mask` is of the shape [H, W]. I am adding the batch dim with `axis=0`

Comment (arashwan) [reply official/vision/beta/projects/panoptic_maskrcnn/dataloaders/panoptic_maskrcnn_input.py:154]: I see, sounds good!

File: official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py (hunk 1 above)
Comment (srihari-humbarwadi) [reply official/vision/beta/projects/panoptic_maskrcnn/configs/panoptic_maskrcnn.py]: done

**Review by arashwan — APPROVED** [review]
